documentation/developers.md

```diff
@@ -30,6 +30,32 @@ Code sample:
       v1-status-callback)
     context))
 ```
+
+
+## Implementing Your Status Function
+The puppetlabs.trapperkeeper.services.status.status-core namespace contains
+some utilities to aid in the implementation of your status functions.  In
+particular, the `level->int` function defines an ordering for status levels as
+`:critical < :info < :debug`, and the `level>`, `level<`, `level>=` and `level<=`
+functions can be used to compare status levels.  These are especially useful in
+conjunction with the `cond->` macro from clojure.core.  Here's an example of how
+a status function might be implemented to utilize the `level>=` function:
+```clj
+(require '[puppetlabs.trapperkeeper.services.status.status-core :as status-core])
+
+(defn my-status
+  [level]
+  (let [level>= (partial status-core/level>= level)]
+    {:is-running true
+     :status (cond-> {:this-is-critical "foo"}
+               (level>= :info) (assoc :bar "bar"
+                                      :baz "baz")
+               (level>= :debug) (assoc :x "x"
+                                       :y "y"
+                                       :z "y"))}))
+```
```

Comment:
Actually, looking at that snippet above, it kind of seems like there is almost never going to be a use case where someone doesn't want to close over a partial function in order to use the `cond->` macro, and if people are already going to be defining that partial function themselves anyway, I'm going back to where I was before where I'm kind of wondering whether or not it's even valuable to provide those utility fns that bake in the comparator.  They don't reduce the lines of code for the consumer, and we agreed we need the generic one anyway, so it seems like they just adds extra LOC and tests to the status library.

If we do include them I'll probably be inclined to use `compare-level` directly in my own consumer code anyway, because it just seems more readable to me.  Since clojure uses prefix notation for everything, `(compare-level >= :info :critical)` seems fairly intuitive to me since the comparison operator is in the "normal" position relative to the two arguments being compared, and it's nice to see that explicitly in my own code.  Whereas with the `level>=` fn, it kind of reads like `(level >= ...)` to me, which then sort of makes the ordering of the two arguments that are being compared ambiguous?

I dunno, maybe that's just me.  I guess it's pretty harmless to leave those extra fns in, but looking at the two code snippets above, I'm pretty sure I'll end up just using the `compare-level` fn directly.